### PR TITLE
Generalize exception type

### PR DESCRIPTION
**Pull request by timholy**
In preparation for https://github.com/JuliaLang/julia/pull/41885

### CODE REVIEW

File: test/test_array.jl [hunk 1]
```diff
@@ -139,7 +139,7 @@ const AG = ArchGDAL;
         AG.readraster("ospy/data4/aster_write.img"; flags = AG.OF_UPDATE) do ds
             @test sprint(print, ds) == """
             GDAL Dataset (Driver: HFA/Erdas Imagine Images (.img))
-            File(s):
```
Comment (yeesian) [inline test/test_array.jl:142]: the tests are whitespace sensitive within the strings, so I suspect this will fail

**Comment by yeesian** [comment]
(It seems that the workflows needed to be manually triggered for PRs by first-time contributors; also sorry for pushing a commit earlier that clashed with your fix and squash)

Thanks!

**Comment by visr** [comment]
> It seems that the workflows needed to be manually triggered for PRs by first-time contributors

Under settings / actions you can change this to be only for new accounts.

**Comment by yeesian** [comment]
Ah thanks! Changed it to be only for new accounts

**Pull request merged**